sormas-ui/src/main/java/de/symeda/sormas/ui/caze/CaseCreateForm.java

```diff
@@ -332,31 +332,46 @@ protected void addFields() {
 			updateFacility();
 		});
 		facilityOrHome.addValueChangeListener(e -> {
-			FieldHelper.removeItems(facilityCombo);
-			if (TypeOfPlace.FACILITY.equals(facilityOrHome.getValue())
-				|| ((facilityOrHome.getValue() instanceof java.util.Set) && TypeOfPlace.FACILITY.equals(facilityOrHome.getNullableValue()))) {
-				if (facilityTypeGroup.getValue() == null) {
-					facilityTypeGroup.setValue(FacilityTypeGroup.MEDICAL_FACILITY);
-				}
-				if (facilityType.getValue() == null && FacilityTypeGroup.MEDICAL_FACILITY.equals(facilityTypeGroup.getValue())) {
-					facilityType.setValue(FacilityType.HOSPITAL);
-				}
 
-				if (facilityType.getValue() != null) {
-					updateFacility();
-				}
+			boolean hasOptionalHealthFacility = UserProvider.getCurrent().hasOptionalHealthFacility();
+			if (hasOptionalHealthFacility) {
+				String facilityId = UserProvider.getCurrent().getUser().getHealthFacility().getUuid();
+				FacilityDto facilityDto = FacadeProvider.getFacilityFacade().getByUuid(facilityId);
+				FacilityType facilityUserType = facilityDto.getType();
```

Comment:
I've also tested another case , changing a Surveillance Officer user role and checking the  "Has optional health facility" checkbox but after logging in with Surveillance Officer user , the same error appeared due to null healthFacility. 
![image](https://user-images.githubusercontent.com/47103965/196446255-f88ebee9-51f8-468f-afaa-c8b0d47ce8d4.png)

![image](https://user-images.githubusercontent.com/47103965/196446832-90de94bb-1a67-49a2-99e5-c25633c8a4ed.png)


![image](https://user-images.githubusercontent.com/47103965/196446307-57590ad1-a14f-4589-8566-ba194fab28b9.png)